From 3a18dbbf7cf95d9d9fbdc38911a755e657fe7695 Mon Sep 17 00:00:00 2001
From: Doug MacEachern <dougm@vmware.com>
Date: Thu, 6 Nov 2014 18:45:34 -0800
Subject: [PATCH] Option to run tests names matching a pattern.

The option can be set using the BATS_TEST_PATTERN env var.

TODO: Add bats cli option
---
 libexec/bats-preprocess | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libexec/bats-preprocess b/libexec/bats-preprocess
index 04297ed..9de5d90 100755
--- a/libexec/bats-preprocess
+++ b/libexec/bats-preprocess
@@ -31,16 +31,26 @@ encode_name() {
 
 tests=()
 index=0
+run_test=1
 pattern='^ *@test  *([^ ].*)  *\{ *(.*)$'
 
 while IFS= read -r line; do
   let index+=1
   if [[ "$line" =~ $pattern ]]; then
     quoted_name="${BASH_REMATCH[1]}"
+    if [ -n "$BATS_TEST_PATTERN" ]; then
+      if [[ "${quoted_name//\"/}" =~ $BATS_TEST_PATTERN ]]; then
+        run_test=1
+      else
+        run_test=0
+      fi
+    fi
     body="${BASH_REMATCH[2]}"
     name="$(eval echo "$quoted_name")"
     encoded_name="$(encode_name "$name")"
-    tests["${#tests[@]}"]="$encoded_name"
+    if [ "$run_test" -eq 1 ]; then
+      tests["${#tests[@]}"]="$encoded_name"
+    fi
     echo "${encoded_name}() { bats_test_begin ${quoted_name} ${index}; ${body}"
   else
     printf "%s\n" "$line"
-- 
2.10.2

